lib/model/CPopulationModel.cc

```diff
@@ -445,16 +445,24 @@ void CPopulationModel::createNewModels(std::size_t n, std::size_t m) {
 void CPopulationModel::updateRecycledModels() {
     CDataGatherer& gatherer = this->dataGatherer();
     for (auto pid : gatherer.recycledPersonIds()) {
-        m_PersonLastBucketTimes[pid] = 0;
+        if (pid < m_PersonLastBucketTimes.size()) {
+            m_PersonLastBucketTimes[pid] = 0;
+        }
     }
 
     TSizeVec& attributes = gatherer.recycledAttributeIds();
     for (auto cid : attributes) {
-        m_AttributeFirstBucketTimes[cid] = CAnomalyDetectorModel::TIME_UNSET;
-        m_AttributeLastBucketTimes[cid] = CAnomalyDetectorModel::TIME_UNSET;
-        m_DistinctPersonCounts[cid] = m_NewDistinctPersonCounts;
-        if (m_NewPersonBucketCounts) {
-            m_PersonAttributeBucketCounts[cid] = *m_NewPersonBucketCounts;
+        if (cid < m_AttributeFirstBucketTimes.size()) {
+            m_AttributeFirstBucketTimes[cid] = CAnomalyDetectorModel::TIME_UNSET;
+            m_AttributeLastBucketTimes[cid] = CAnomalyDetectorModel::TIME_UNSET;
+            m_DistinctPersonCounts[cid] = m_NewDistinctPersonCounts;
+            if (m_NewPersonBucketCounts) {
+                m_PersonAttributeBucketCounts[cid] = *m_NewPersonBucketCounts;
+            }
+        } else {
+            LOG_ERROR("Recycled attribute identifier '"
```

Comment:
There's a missing `<<` after the `LOG_ERROR(`.

Also, have you run `clang-format` on this file?  CI is set to fail now if the formatting isn't what `clang-format` would choose.